components/Feature/VulnerabilitiesGrid/ResourceCard/index.js

```diff
@@ -123,7 +145,7 @@ const ResourceCard = ({
         </div>
         {fullDescription && (
           <p>
-            {first}
+            {getHighlighted(first)}
```

Comment:
Related to the previous comment:

I was wondering whether we could have made our own bold symbols decoder, which would be safer.
So say we have our description text: `It's better to have hammered-in screw than to have screwed-in nail.` and our search text: `Hammer`. So we would encode our description text with some arbitrary markings like, say `_#word#_` to signify parts that are meant to be highlighted. We would get this encoded text as a result: `It's better to have _#hammer#_ed-in screw than to have screwed-in nail.`

Then in this place here rather than doing `<p>{somefunction(description)} ...`, we could instead do something like this:
`description.split('_').map(snippet => /#[^#]+#/g.test(snippet) ? <p className='our-highlight-class'>{snippet.replace(/#(.+)#/, "$1")}<p> : <p>{snippet}<p>);` _(this could be extracted into a function)_

Which would be both safer & would allow us to control highlighting style through CSS modules or something. Effectively, this would split any text from a single paragraph into multiple paragraphs & the text with bold markings would be a paragraph with special class name.

Reply:
I might be misunderstanding, but in the end this method would also find and replace the text with the same text but enclosed within a <p></p>?
Seems like the same implementation but with an extra step to add _# #_ ?
We can also likely add our own css by using <p></p> with a className instead of <mark>, <mark> was just the default option and it's quite intuitive (might be worth checking if we have any designs on this)

Reply:
Me and @kosiakkatrina talked this one through on Tandem. Conclusion is that getting this to work requires more code changes on the `"full phrase highlighting"` side of code. So we're starting to look at a refactoring here, which won't be tackled on now.